modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java

```diff
@@ -1472,6 +1477,32 @@ private static void notifyStateChange(@Nullable String igniteInstanceName, Ignit
             lsnr.onStateChange(igniteInstanceName, state);
     }
 
+    /**
+     * @return map of resources for overriding and clear the map after it.
+     */
+    public static Map<String, WrappableResource> takeTestResources() {
```

Comment:
It's a very tricky semantics: resources will be injected only to next node started by current thread. 
It becomes hardly usable if we want to start several nodes via startGrids(N). Let's give user control for clearing the resources:

IgnitionEx.addTestResource(...);

try {
    startGrids(N);
}
finally {
    IgnitionEx.clearTestResources();
}

The same approach is assumed for IgnitionEx#setClientMode.